frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_user_view.dart

```diff
@@ -49,28 +50,84 @@ class SettingsUserView extends StatelessWidget {
             crossAxisAlignment: CrossAxisAlignment.start,
             mainAxisSize: MainAxisSize.min,
             children: [
-              _renderUserNameInput(context),
-
+              _buildUserIconSetting(context),
               if (isSupabaseEnabled) ...[
-                const VSpace(20),
+                const VSpace(12),
                 UserEmailInput(user.email)
               ],
-
-              const VSpace(20),
-              _renderCurrentIcon(context),
-              const VSpace(20),
+              const VSpace(12),
               _renderCurrentOpenaiKey(context),
-              const VSpace(20),
-              // _renderHistoricalUser(context),
+              const VSpace(12),
               _renderLoginOrLogoutButton(context, state),
-              const VSpace(20),
+              const VSpace(12),
             ],
           ),
         ),
       ),
     );
   }
 
+  Row _buildUserIconSetting(BuildContext context) {
+    return Row(
+      children: [
+        UserAvatar(
+          iconUrl: user.iconUrl,
+          name: user.name,
+          isLarge: true,
+        ),
+        const HSpace(8),
+        Column(
+          children: [
+            FlowyIconButton(
+              width: 24,
+              tooltipText: LocaleKeys.settings_user_tooltipRemoveIcon.tr(),
+              icon: const FlowySvg(FlowySvgs.restore_s),
+              onPressed: () => context
+                  .read<SettingsUserViewBloc>()
+                  .add(const SettingsUserEvent.updateUserIcon()),
```

Comment:
nit: I think it's a bit untuitive from readability standpoint that sending this update event with no arguments should clear it. I would prefer ""

Reply:
I'll change it, however, I did this because I think the act of removing the icon by sending an empty string is the _unintuitive_ part.

Reply:
Gotcha, I see what you mean. In that case,  how about a separate SettingsUserEvent.restoreUserIcon?

Reply:
Yeah, or `removeUserIcon`, I'll implement that instead.